src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/SpinWaitScenario.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTestApp.Scenarios
+{
+    /// <summary>
+    /// Synchronously spins until it receives the Continue command.
+    /// </summary>
+    internal class SpinWaitScenario
+    {
+        public static Command Command()
+        {
+            Command command = new(TestAppScenarios.SpinWait.Name);
+            command.Handler = CommandHandler.Create((Func<CancellationToken, Task<int>>)ExecuteAsync);
+            return command;
+        }
+
+        public static Task<int> ExecuteAsync(CancellationToken token)
+        {
+            return ScenarioHelpers.RunScenarioAsync(async logger =>
+            {
+                await ScenarioHelpers.WaitForCommandAsync(TestAppScenarios.SpinWait.Commands.StartSpin, logger);
+
+                Task continueTask = Task.Run(() => ScenarioHelpers.WaitForCommandAsync(TestAppScenarios.SpinWait.Commands.StopSpin, logger));
+
+                while (!continueTask.IsCompleted)
```

Comment:
Nit: could you achieve the same thing with SpinWait.SpinUntil

Reply:
I'll have to see if the produces enough detectable CPU usage, but I'll give it a try.

Reply:
The SpinWait class yields the time slice of the current thread to other threads after a short time, so we end up using very little CPU. I've switched to using Thread.SpinWait which will utilize CPU on the thread for a number of "iterations", which is processor speed dependent. I chose an arbitrarily large number such that it doesn't constantly check the completion of the task; an iteration value of 1,000,000 seems to be about 100-150 ms on my Surface Book 2 when running on battery.